src/main/resources/assets/js/components/AppSearchComponent.jsx

```diff
@@ -0,0 +1,52 @@
+/** jsx React.DOM */
+
+var React = require("react/addons");
+
+var AppSearchComponent = React.createClass({
+  displayName: "AppSearchComponent",
+
+  propTypes: {
+    onSearch: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      searchValue: null
+    };
+  },
+
+  handleChange: function (event) {
+    this.setState({searchValue: event.target.value});
+    this.search(event.target.value);
+  },
+
+  handleSubmit: function (event) {
+    event.preventDefault();
+  },
+
+  search: function (value) {
+    this.props.onSearch(value);
+  },
+
+  resetSearch: function () {
+    this.setState({searchValue: null});
+    this.props.onSearch();
+  },
+
+  render: function () {
+    var inputClass = React.addons.classSet({
+      "input-success": this.state.searchValue,
+      "form-control": true
+    });
+
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <input className={inputClass} type="text" value={this.state.searchValue}
```

Comment:
Create a ref here to access the element in a handler on anywhere else in this component.
```
<input ref="searchInput" ... />
```